src/CoreBundle/Form/Type/BooleanType.php

```diff
@@ -19,16 +19,13 @@
     );
 }
 
-class_alias(
-    \Sonata\Form\Type\BooleanType::class,
-    __NAMESPACE__.'\BooleanType'
-);
-
-if (false) {
-    /**
-     * @deprecated Since version 3.x, to be removed in 4.0.
-     */
-    class BooleanType extends \Sonata\Form\Type\BooleanType
+/**
+ * @deprecated Since version 3.x, to be removed in 4.0.
+ */
+class BooleanType extends \Sonata\Form\Type\BooleanType
```

Comment:
This looks like a BC-break: a class type hinting against the old type but receiving the new type will not be able to work properly.

Reply:
If you have any class anywhere that type hints against `\Sonata\CoreBundle\Form\Type\BooleanType`, it will not be happy when receiving a `\Sonata\Form\Type\BooleanType` . But maybe we can agree this is probably not a big deal for forms?

Reply:
IMHO, that's the expected behavior.

If you have something that needs the old component, everything will work as before. If you explicitly want the new stuff, you can use the new component without any problems.

Reply:
> If you have something that needs the old component, everything will work as before. If you explicitly want the new stuff, you can use the new component without any problems.

Is that correct? This crashes, right?

```php

public class Foo
{
    // something that needs the old component
    public function __construct(\Sonata\CoreBundle\FormType\BooleanType $bar)
    {
    }
}
new Foo(new \Sonata\Form\Type\BooleanType(…$args));
```

Reply:
That's correct, but the expected behaviour.

If you have something that needs an A, you can't pass an B to it. You need to adopt the component that needs an A too.

Reply:
Ok. I think you need to remove this: https://github.com/sonata-project/SonataCoreBundle/blob/3.x/src/Form/Type/BooleanType.php#L102 for all forms then. Please note that this has happened in cases where the something that needs an A is out of your control, for instance in the case of https://github.com/sonata-project/SonataCoreBundle/issues/618

Reply:
Fixed it